### PR TITLE
Fix GEOS-Chem 14.5.0 build in CESM by excluding hard dependencies on HEMCO

### PR DESCRIPTION
**Pull request by jimmielin**
### Name and Institution (Required)

Name: Haipeng Lin
Institution: Harvard Univ.

### Describe the update

In the CESM model, emissions through HEMCO are provided to GEOS-Chem through the physics buffer and not through the HEMCO subroutine calls. A side-effect of this implementation is that HEMCO versions and GEOS-Chem versions in CESM can be decoupled, and thus hard-dependencies on HEMCO data structures should not apply.

In this particular case, additional ExtState fields in HEMCO are causing CESM build failures in the latest GEOS-Chem version. Most (but not all) subroutines in hco_interface_gc_mod.F90 are unused in CESM and are now blocked off from MODEL_CESM to avoid this hard dependency.

Note: `HCOI_GC_WriteDiagn` was rearranged in the module but otherwise unchanged, in order to use one `#if !defined( MODEL_CESM )`

### Expected changes

None

### Reference(s)

N/A

### Related Github Issue

Building CESM with latest GEOS-Chem tag `14.5.0-rc.0`

### CODE REVIEW

**Review by yantosca — CHANGES_REQUESTED** [review]
Looks good @jimmielin.  We just need a changelog update.

**Comment by jimmielin** [comment]
Thanks @yantosca, changelog updated!

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Removed EDGAR8_CH4_AWB emissions from CH4 and carbon simulations to avoid double counting with GFED
 - Fixed formatting error in `.github/workflows/stale.yml` that caused the Mark Stale Issues action not to run
 - Fixed emissions in GCHP carbon ExtData.rc so that data in molecules/cm2/s are converted to kg/m2/s
+- Excluded HEMCO interface and ExtState fields from `MODEL_CESM` in `hco_interface_gc_mod.F90` for compatibility with CESM, which runs HEMCO separately
```
Comment (lizziel) [inline CHANGELOG.md:43]: Could you move this to Changed? Fixed implies there was a bug in a previous version which in this case I don't think there was.

Comment (jimmielin) [reply CHANGELOG.md]: Yes, done! Thank you

**Review by yantosca — APPROVED** [review]
Looks good!

**Comment by msulprizio** [comment]
Note that I have changed the base branch to `dev/no-diff-to-benchmark` since this will go into 14.5.1 and not 14.5.0.

**Comment by lizziel** [comment]
@yantosca, looks like this branch has no merge conflicts so should be good to merge.

**Comment by yantosca** [comment]
All GEOS-Chem Classic integration tests passed, except for tagCO (known restart file issue)
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

CodeDir   : a8aeb02 GEOS-Chem update: Merge PR #2578 (Fixes for GCHP adjoint )
GEOS-Chem : 62f3db08b Merge PR #2578 (Fixes to compile GCHP adjoint)
HEMCO     : deaa192 HEMCO 3.10.0 release
Cloud-J   : f8a2b7f Update version number for 8.0.1 release
HETP      : 2a99b24 Merge pull request #2 from geoschem/bugfix/initialize_local_variables

Using 24 OpenMP threads
Number of execution tests: 30

Submitted as SLURM job: 59103476
==============================================================================

...
gc_4x5_merra2_tagCO.................................Execute Simulation....FAIL
...
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 29
Execution tests failed: 1
Execution tests not yet completed: 0
```
Also all tests were zero-diff w/r/t the previous integration test except for:
- TOMAS (parallelization issue?)
- APM (parallelization issue?)

**Comment by yantosca** [comment]
All GCHP integration tests passed:
```console
==============================================================================
GCHP: Execution Test Results

CodeDir       : 7c2d8e2 GEOS-Chem update: Merge PR #2578 (Fixes for GCHP adjoint )
MAPL          : 9ad63ae Merge PR #37 containing update to vertically flip imports with dimensionless pressure proxy lev coordinates
GMAO_Shared   : 4ddb3ec Merge pull request #2 from geoschem/feature/mapl-upgrade
ESMA_cmake    : ad5deba Added ecbuild as a submodule of ESMA_cmake
gFTL-shared   : 4b82492 Merge branch 'upstream_v1.5.0' into feature/v1.5.0
FMS           : 259759d Merge pull request #3 from geoschem/feature/update_gmao_libs
FVdycoreCubed : af42462 Merge PR #8 (Add PLEadv diagnostic for offline advection in GCHP)
geos-chem     : 62f3db08b Merge PR #2578 (Fixes to compile GCHP adjoint)
HEMCO         : deaa192 HEMCO 3.10.0 release
yaFyaml       : 19afe50 Merge branch 'upstream_v1.0.4' into feature/v1.0.4
pFlogger      : 2c4b724 Merge branch 'upstream_v1.9.1' into feature/v1.9.1
Cloud-J       : f8a2b7f Update version number for 8.0.1 release
HETP          : 2a99b24 Merge pull request #2 from geoschem/bugfix/initialize_local_variables

Number of execution tests: 12

Submitted as SLURM job: 59103750
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
Also, all tests were zero-diff w/r/t the previous set of integration tests.
